src/api/photoApiClient.js

```diff
@@ -0,0 +1,74 @@
+import request from 'request';
+import API_KEY from './apiKey';
+import RoverPhoto from '../models/roverPhoto';
+import Rover from '../models/rover';
+
+export default function photos() {
+  const curiosityCameras = ['FHAZ', 'RHAZ', 'MAST', 'CHEMCAM', 'MAHLI', 'MARDI', 'NAVCAM'];
+  const spiritCameras = ['FHAZ', 'RHAZ', 'NAVCAM', 'PANCAM', 'MINITES'];
+  const opportunityCameras = ['FHAZ', 'RHAZ', 'NAVCAM', 'PANCAM', 'MINITES'];
+
+  const rovers = [new Rover('curiosity', curiosityCameras),
+    new Rover('spirit', spiritCameras),
+    new Rover('opportunity', opportunityCameras)];
+
+  const output = rovers.map(rover =>
+    getManifest(rover.name)
+      .then(manifest => getSols(manifest, rover))
+      .then(sols => getPhotos(rover, sols)));
+
+  return output;
+}
+
+
+function getManifest(rover) {
+  return new Promise((resolve, reject) => {
+    const url = `https://api.nasa.gov/mars-photos/api/v1/manifests/${rover}?&api_key=${API_KEY}`;
+
+    request(url, (error, response, body) => {
+      if (response.statusCode !== 200) {
+        reject(Error(`failed to get manifest from ${rover}`));
+      } else {
+        const manifest = JSON.parse(body);
+        resolve(manifest);
+      }
+    });
+  });
+}
+
+function getSols(manifest, rover) {
+  return new Promise(resolve => {
+    const sols = rover.cameras.map(camera => {
+      let maxSol;
+      let photoManifest = manifest.photo_manifest.photos;
+      let i = photoManifest.length;
+      while (!maxSol && i > 0) {
+        i -= 1;
+        maxSol = photoManifest[i].cameras.includes(camera.shortName) ? photoManifest[i].sol : null;
+      }
+      return maxSol;
+    });
+    resolve(sols);
+  });
+}
+
+
+function getPhotos(rover, sols) {
```

Comment:
I think this could have a better name - it's returning a Rover so `getPhotos` is a bit misleading.

Maybe `getRoversWithPhotos`?